utils/types/reporting_types.go

```diff
@@ -35,6 +35,14 @@ var (
 	WAREHOUSE              = "warehouse"
 )
 
+var (
+	// Tracking plan validation states
+	SUCCEEDED                    = "succeeded"
+	SUCCEEDED_WITHOUT_VIOLATIONS = "succeeded_without_violations"
```

Comment:
In trackingplan stage, there will be no status SUCCEEDED unlike other stages.
`SUCCEEDED` is split into `SUCCEEDED_WITHOUT_VIOLATIONS` + `SUCCEEDED_WITH_VIOLATIONS`

Reply:
Can't we skip `SUCCEEDED_WITHOUT_VIOLATIONS`? If it succeeded, it succeeded. If there are violations, I would also skip the `SUCCEEDED` part and use something more straightforward, e.g. `VIOLATIONS_FOUND`, since what you are interested in is mostly that violations were found, not that we are ignoring these violations by considering the events as succeeded and letting them flow further in the pipeline.

Reply:
All events sent to tracking plan stage can be categorized into 3 disjoint sets `SUCCEEDED_WITHOUT_VIOLATIONS` + `SUCCEEDED_WITH_VIOLATIONS` + `ABORTED (due to violations)`. User is more interested in `VIOLATIONS_FOUND (SUCCEEDED_WITH_VIOLATIONS+ ABORTED)` and `ABORTED`.Also User might ask for only succeeded without violations. In view of this, 3 disjoint status data are sent to reporting, graphs shown in UI can be constructed based on user requirements

Reply:
Letting events succeed with violations is a configuration setting given to user. User might this setting always without dropping events.